File: Domain/Reserva.cs
using System;
using Domain.Enums;

namespace Domain
{
    public class Reserva
    {
     public int Id { get; set; }
     public string NombreTurista { get; set; }
     public string ApellidoTurista { get; set; }
     public string MailTurista { get; set; }
     public DateTime CheckIn { get; set; }
     public DateTime CheckOut { get; set; }
     public int CantidadHuespedes { get; set; }
     public virtual EstadoReserva Estado { get; set; }
     public virtual Hospedaje Hospedaje { get; set; }
     public string Descripcion { get; set; }
    }
}

File: SPA/FrontEnd/src/app/Models/Reserva/reserva.ts
import { Hospedaje } from '../hospedaje/hospedaje';
import { EstadoReserva } from '../EstadoReserva/estado-reserva.enum';
export class Reserva {
    id: number;
    nombreTurista: string;
    apellidoTurista: string;
    mailTurista: string;
    checkIn: Date;
    checkOut: Date;
    cantidadHuespedes: number;
    estado: EstadoReserva;
    hospedaje: Hospedaje;
    descripcion: string;
    constructor(id: number, desc: string, hospedaje: Hospedaje, estado: EstadoReserva,
        mailTurista: string, apellidoTurista: string, nombreTurista: string) {
            
        this.id = id;
        this.descripcion = desc;
        this.hospedaje = hospedaje;
        this.estado = estado;
        this.mailTurista = mailTurista;
        this.nombreTurista = nombreTurista;
        this.apellidoTurista = apellidoTurista;
    }
}